apps/site/src/pages/api/verify-email.api.ts

```diff
@@ -78,7 +88,28 @@ export default createBaseHandler<
         );
       }
 
-      await user.update(db, { hasVerifiedEmail: true });
+      if (emailVerificationCode.variant === "linkWordpress") {
+        const { wordpressInstanceUrl } = emailVerificationCode;
+        if (!wordpressInstanceUrl) {
```

Comment:
That's fair. I don't think we should do it in this PR to keep the scope fairly small. I'm not sure if we have a task for something like this, but it would be ideal if it's some established pattern that we colocate things that are related instead of relying on runtime checks for optional values.